Makefile

```diff
@@ -68,3 +74,30 @@ minikube-rbac-test:
 
 fmt:
 	$(GOFMT) -s -w $(GO_FILES)
+
+
+HAS_GOX         := $(shell which gox)
+HAS_GOLINT      := $(shell which golint)
+HAS_KUBECFG     := $(shell which kubecfg)
+
+bats:
+	git clone --depth=1 https://github.com/sstephenson/bats.git
+
+ksonnet-lib:
+	git clone --depth=1 https://github.com/ksonnet/ksonnet-lib.git
+
+.PHONY: bootstrap
+bootstrap: bats ksonnet-lib
+
+ifndef HAS_GOX
```

Comment:
An alternative would be to run `go get ...` unconditionally, since it won't re-download the tool if it already exists.  The difference vs your explicit which-conditional approach is that go-get would force the binary to exist in $GOPATH/bin, whereas yours won't re-download if the tool exists elsewhere in $PATH.

Yet another alternative would be to do the conditional in shell, rather than make.  (`if ! which kubecfg >/dev/null; do wget ... ; done`)

I would probably pick one of these rather than the more verbose make conditionals - but it's totally a personal style / more-tightly-contained thing.  The current version is fine by me too.

Reply:
I wasn't 100% sure with `got get ...` no re-download so I was more on the cautious side. It makes sense to simply fetch the tools. I would be very unlikely that these tools are installed anywhere else in $PATH. 

I did the changes and like the less verbose version.